backend/src/routes/api/status/statusUtils.ts

```diff
@@ -0,0 +1,133 @@
+import { FastifyRequest } from 'fastify';
+import { KubeFastifyInstance, KubeStatus } from '../../../types';
+import createError from 'http-errors';
+import { CoreV1Api, CustomObjectsApi } from '@kubernetes/client-node';
+
+type groupObjResponse = {
+  users: string[] | null;
+};
+
+const SYSTEM_AUTHENTICATED = 'system:authenticated';
+const GROUPS_CONFIGMAP_NAME = 'groups-config';
+const DEFAULT_USERNAME = 'kube:admin';
+
+export class ConfigGroupError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, ConfigGroupError.prototype);
+  }
+
+  getErrorMessage(): string {
+    return 'Error retrieving ConfigMap ' + this.message;
+  }
+}
+
+export class GroupError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, GroupError.prototype);
+  }
+
+  getErrorMessage(): string {
+    return 'Error retrieving Group ' + this.message;
+  }
+}
+
+export const status = async (
+  fastify: KubeFastifyInstance,
+  request: FastifyRequest,
+): Promise<{ kube: KubeStatus }> => {
+  const kubeContext = fastify.kube.currentContext;
+  const { currentContext, namespace, currentUser, clusterID, clusterBranding } = fastify.kube;
+  const currentUserName =
+    (request.headers['x-forwarded-user'] as string) || currentUser.username || currentUser.name;
+  let userName = currentUserName?.split('/')[0];
+  if (!userName || userName === 'inClusterUser') {
+    userName = DEFAULT_USERNAME;
+  }
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  const coreV1Api = fastify.kube.coreV1Api;
+  let isAdmin = false;
+
+  try {
+    const groupConfig = await getGroupsConfig(coreV1Api, namespace);
+    const adminGroup = await getGroupsAdminConfig(coreV1Api, namespace, groupConfig);
+
+    if (adminGroup === SYSTEM_AUTHENTICATED) {
+      isAdmin = true;
+    } else {
+      const adminUsers = await getGroup(customObjectsApi, adminGroup);
+      isAdmin = adminUsers?.includes(userName) ?? false;
+    }
+  } catch (e) {
+    fastify.log.error(e.getErrorMessage());
```

Comment:
should the error message appear also when the user group (no admin group) doesn't exist?

Reply:
Yeah, that's what is doing, if the group doesn't exist it will trigger the error, the only time it won't happen would be when the group is empty. (as it shouldn't be an error)

Reply:
it looks to me like the try/catch is only applied to the process for fecthing the admin-groups (line 59), I expected to see the same done for users group (i.e., allowed-groups). But maybe I'm missing something, first time seeing this code